src/main/cli/commandLine.ts

```diff
@@ -0,0 +1,92 @@
+import { PublicationRepository } from "readium-desktop/main/db/repository/publication";
+import { container } from "readium-desktop/main/di";
+import { RootState } from "readium-desktop/main/redux/states";
+import { CatalogService } from "readium-desktop/main/services/catalog";
+import { Store } from "redux";
+import { URL } from "url";
+import { isArray } from "util";
+
+export async function cli_(filePath: string[] | string) {
+
+    let publicationOpenRequested = false;
+    let returnValue = true;
+    const filePathArray = isArray(filePath) ? filePath : [filePath];
+
+    for (const fp of filePathArray) {
+        // import and read publication
+        const catalogService = container.get("catalog-service") as CatalogService;
+        const publication = await catalogService.importFile(fp);
+        const store = container.get("store") as Store<RootState>;
+        if (publication) {
+            if (!publicationOpenRequested) {
+                store.dispatch({
+                    type: readerActions.ActionType.OpenRequest,
+                    payload: {
+                        publication: {
+                            identifier: publication.identifier,
+                        },
+                    },
+                });
+                publicationOpenRequested = true;
+            }
+        } else {
+            process.stderr.write(`Publication error for "${filePath}"\n`);
+            returnValue = false;
+        }
+    }
+    return returnValue;
+}
+
+export async function cliImport(filePath: string) {
+    try {
+        const catalogService = container.get("catalog-service") as CatalogService;
+        if (!await catalogService.importFile(filePath)) {
+            throw new Error();
+        }
+    } catch (e) {
+        return false;
+    }
+    return true;
+}
+
+export async function cliOpds(title: string, url: string) {
+    // extract and save the title and url from opdsFeed
+    // title=http://myurl.com or get TLD and set url
+    const hostname = (new URL(url)).hostname;
+    if (hostname) {
+        const opdsRepository = container.get("opds-feed-repository") as OpdsFeedRepository;
+        await opdsRepository.save({ title, url });
+        return true;
+    }
+    return false;
+}
+
+export async function cliRead(title: string) {
+    // read the publication name
+    const publicationRepo = container.get("publication-repository") as PublicationRepository;
+    const publication = await publicationRepo.searchByTitle(title);
+    if (publication && publication.length) {
+        const store = container.get("store") as Store<RootState>;
+        store.dispatch({
+            type: readerActions.ActionType.OpenRequest,
+            payload: {
+                publication: {
+                    identifier: publication[0].identifier,
+                },
+            },
+        });
+    } else {
+        process.stdout.write(`There is no publication title match for "${title}"\n`);
```

Comment:
Use `process.stderr` instead of `process.stdout` (error condition, exist status code == 1)

Reply:
done

Reply:
done